### ISSUE TITLE
Proposal: prioritize migration for customer-facing code

### ISSUE THREAD

**Issue by nate-thegrate** [flutter/flutter#151373]
The current state of the [`Checkbox` API docs](https://api.flutter.dev/flutter/material/Checkbox-class.html#longSnippet1):

[![checkbox docs](https://github.com/flutter/flutter/assets/10457200/7649091d-569e-494c-ac81-f925039e2e4b)](https://api.flutter.dev/flutter/material/Checkbox-class.html#longSnippet1)

In some sense, every part of an open source project is "customer-facing", but I think it's especially important for links & examples shown on [api.flutter.dev](https://api.flutter.dev/) to be up-to-date.

**Comment by cedvdb** [comment]
I feel like that should have been caught somewhere when the deprecation has been introduced to begin with, shouldn't it ?

**Comment by nate-thegrate** [comment]
The doc comment analysis is smart enough to trigger warnings, but currently we ignore deprecated member use, based on #143312.

https://github.com/flutter/flutter/blob/fd3f769ec5ba75e977a5039eb8cc537061d44294/analysis_options.yaml#L23-L27

One way to mitigate this problem would be to update the [example code analysis options](https://github.com/flutter/flutter/blob/master/examples/api/analysis_options.yaml) so it notices deprecated member use, though that would make big PRs like #149963 ($\textcolor{green}{\textsf{+1,194 }}\textcolor{#d1242f}{\textsf{−537}}$ diffs) even bigger.

**Comment by MitchellGoodwin** [comment]
Yes, we typically don't jump to remove all uses of deprecated API from the docs until we fully remove it, or it seems somewhat stable. When it's only used in one place in a widget it also makes a decent artifact when we are in the in-between period between marking something as deprecated and actually removing it. Which is normally not a problem, but `MaterialState` was used in some widgets so much that it looks really ugly like this.

I don't think we want to have the examples care more strongly about deprecated member use, because it complicates the already awkward deprecation cycle, by having things fail all over the place when we first mark something as deprecated.
